### PR TITLE
Bump UniFFI to pre `0.29.1`

### PR DESCRIPTION
**Pull request by CyonAlexRDX**
> [!NOTE]
> Target branch is not `main`
> We should merge @micbakos-rdx PR #288 before we merge this

Bump UniFFI from `0.27.1` -> `0.28.3+` (pre `0.29.X`) specifically [the latest commit on UniFFI main branch](https://github.com/mozilla/uniffi-rs/commit/2c003b16d1e70e1914b5d8ceb517eef3676cd187)

## New system for `custom_types`
The big change is [documented here](https://mozilla.github.io/uniffi-rs/next/Upgrading.html)

**Before**
```rust
impl UniffiCustomTypeConverter for NewCustomType {
    type Builtin = BridgeType;

    fn into_custom(val: Self::Builtin) -> uniffi::Result<Self> {
        ...
    }

    fn from_custom(obj: Self) -> Self::Builtin {
        ...
    }
}
```

**After**
```rust
uniffi::custom_type!(NewCustomType, BridgeType, {
    try_lift: |val| { Ok(...) },
    lower: |obj| { ... },
})
```

### CODE REVIEW

File: crates/sargon/src/signing/testing/simulated_user.rs [hunk 1]
```diff
@@ -9,15 +9,13 @@ pub(crate) enum SigningUserInput {
     Reject,
 }
 
+pub type SpyOnRequest<ID> =
```
Comment (CyonAlexRDX) [inline crates/sargon/src/signing/testing/simulated_user.rs:12]: Just clippy fixes on top of @micbakos-rdx changes

File: crates/sargon/src/signing/testing/simulated_user.rs [hunk 2]
```diff
@@ -9,15 +9,13 @@ pub(crate) enum SigningUserInput {
...
+    Arc<dyn Fn(FactorSourceKind, IndexSet<InvalidTransactionIfNeglected<ID>>)>;
```
Comment (micbakos-rdx) [inline crates/sargon/src/signing/testing/simulated_user.rs:13]: 👌

File: crates/sargon/src/signing/collector/signatures_collector.rs [hunk 3]
```diff
@@ -45,8 +45,7 @@ impl<S: Signable> SignaturesCollector<S> {
     }
 
     pub async fn collect_signatures(self) -> Result<SignaturesOutcome<S::ID>> {
-        self
-            .sign_with_factors() // in decreasing "friction order"
+        self.sign_with_factors() // in decreasing "friction order"
```
Comment (micbakos-rdx) [inline crates/sargon/src/signing/collector/signatures_collector.rs:48]: When I did this change, swift format changed it back to two lines. I don't understand why.

**Review by micbakos-rdx — APPROVED** [review]

File: crates/sargon/src/signing/collector/signatures_collector.rs (hunk 3 above)
Comment (CyonAlexRDX) [reply crates/sargon/src/signing/collector/signatures_collector.rs:48]: did `swiftformat` do that? Swift? Not Cargo fmt?

Comment (micbakos-rdx) [reply crates/sargon/src/signing/collector/signatures_collector.rs:48]: Meant fmt, sorry, force of recent habit

Comment (CyonAlexRDX) [reply crates/sargon/src/signing/collector/signatures_collector.rs:48]: ok strange, we can look into it, that `clippy` and `fmt` should both use same rules...

Comment (micbakos-rdx) [reply crates/sargon/src/signing/collector/signatures_collector.rs:48]: Although earlier this method looked like this:

```
let _ = self
            .sign_with_factors()
```

Comment (CyonAlexRDX) [reply crates/sargon/src/signing/collector/signatures_collector.rs:48]: the `let _ = ` is not really needed I guess

**Review by GhenadieVP — APPROVED** [review]

**Comment by CyonAlexRDX** [comment]
swift tests broke. needs to fix

**Comment by CyonAlexRDX** [comment]
Bump UniFFI to 0.28.3 (pre 0.29.X)

**Pull request merged**